scripts/combine/setupCombineWMass.py

```diff
@@ -59,11 +60,23 @@ def main(args,xnorm=False):
     
     datagroups = make_datagroups_2016(args.inputFile, excludeGroups=excludeGroup, filterGroups=filterGroup, applySelection= not xnorm)
 
-    if args.xlim:
-        if len(args.fitvar.split("-")) > 1:
-            raise ValueError("Restricting the x axis not supported for 2D hist")
-        s = hist.tag.Slicer()
-        datagroups.setGlobalAction(lambda h: h[{args.fitvar : s[complex(0, args.xlim[0]):complex(0, args.xlim[1])]}])
+    if args.axlim or args.rebin:
```

Comment:
Nice feature, does it also work for gen bins? I would say so as long as they are encoded in axes for signal histograms. 
It's probably not optimal to make histogram with granular gen binning if one expects to rebin later, but for some particular studies (e.g. stat only or with few systematics) it might be convenient since remaking histograms take some time.

Reply:
As it's set up now, I don't think it would work for the gen axis, since it reuses the fitvar variable (the axes that are kept in the hist) to align axes with their new binning. To support gen axis rebinding, we need a new rebinAxes variable, which could default to the value of fitvar. I can implement this if you want

Reply:
As you prefer, but probably better to postpone that for now. It may actually be better to simply generate histograms with the desired gen-level binning, even if just for tests.